sources/logdog/src/exec_to_file.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
+use std::process::{Command, Stdio};
+
+use crate::error::{FileError, IoError, Result};
+use snafu::ResultExt;
+
+/// Aggregates the information needed to run a shell command and pipe it to a file.
+#[derive(Debug, Clone)]
+pub(crate) struct ExecToFile {
+    pub command: &'static str,
+    pub args: Vec<&'static str>,
+    pub output_filename: &'static str,
+}
+
+impl ExecToFile {
+    /// Runs a shell command and pipes its output to a named file in the specified outdir.
+    pub(crate) fn run(&self, outdir: &PathBuf) -> Result<()> {
+        let opath = outdir.join(self.output_filename);
+        let mut ofile = File::create(&opath).context(FileError {
+            path: opath.clone(),
+        })?;
+        let efile = ofile.try_clone().context(FileError {
+            path: opath.clone(),
+        })?;
+        ofile
+            .write(format!("{:?}\n", self).into_bytes().as_slice())
+            .context(FileError { path: opath })?;
```

Comment:
Also, separately, `write!` takes away some of those hassles.
```suggestion
        write!(&mut ofile, "{:?}\n", self)
            .context(FileError { path: opath })?;
```
You can use this pattern in other places where you're currently doing a `file.write(format!(`